src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -177,17 +192,7 @@ public void traverse() {
             // prepare the iterator
             iterator = (hasIntervals()) ? new IntervalOverlappingIterator<>(libs, intervalsForTraversal, header.getSequenceDictionary()) : libs;
         }
-
-        // iterate over each alignment, and apply the function
-        final Spliterator<AlignmentContext> spliterator = Spliterators.spliteratorUnknownSize(iterator, 0);
-        StreamSupport.stream(spliterator, false)
-            .forEach(alignmentContext -> {
-                        final SimpleInterval alignmentInterval = new SimpleInterval(alignmentContext);
-                        apply(alignmentContext, new ReferenceContext(reference, alignmentInterval), new FeatureContext(features, alignmentInterval));
-                        progressMeter.update(alignmentInterval);
-                }
-            );
-        logger.info(countedFilter.getSummaryLine());
+        return iterator;
```

Comment:
It does not seem to be a need to keep the return statement at the end, you could have a return in conditional block. In any case if you keep it like this then make the iterator variable final.